### PR TITLE
State expiration renames 

### PR DESCRIPTION
**Pull request by sisuresh**
### What

[TODO: Short statement about what is changing.]

### Why

[TODO: Why this change is being made. Include any context required to understand the why.]

### Known limitations

[TODO or N/A]

### CODE REVIEW

File: soroban-env-common/env.json [hunk 1]
```diff
@@ -1373,52 +1373,52 @@
...
                         },
                         {
-                            "name": "high_expiration_watermark",
+                            "name": "extend_to",
                             "type": "U32Val"
                         }
                     ],
                     "return": "Void",
-                    "docs": "If the entry expiration is below `low_expiration_watermark` ledgers from the current ledger (inclusive), then bump the expiration to be `high_expiration_watermark` from the current ledger (inclusive)"
+                    "docs": "If the entry expiration is below `threshold` ledgers from the current ledger (inclusive), then bump the expiration to be `extend_to` from the current ledger (inclusive)"
```
Comment (SirTyson) [inline soroban-env-common/env.json:1385]: Remove the term "expiration" and "bump" from the docs. Maybe something like "If the entry's TTL is below `threshold` ledgers, extend `liveUntilLedger` such that `TTL == extend_to`, where TTL is defined as `liveUntilLedger - currentLedger`. I think we should definitely expose the term TTL in the SDK, but this doc may be too low level for the term, I'm not sure.

File: soroban-env-common/env.json [hunk 2]
```diff
@@ -1373,52 +1373,52 @@
...
                         },
                         {
-                            "name": "high_expiration_watermark",
+                            "name": "extend_to",
                             "type": "U32Val"
                         }               
                     ],
                     "return": "Void",
-                    "docs": "If expiration of the provided contract instance and code (if applicable) is below `low_expiration_watermark` ledgers from the current ledger (inclusive), then bump the expiration to be `high_expiration_watermark` from the current ledger (inclusive)"
+                    "docs": "If expiration of the provided contract instance and code (if applicable) is below `threshold` ledgers from the current ledger (inclusive), then bump the expiration to be `extend_to` from the current ledger (inclusive)"
```
Comment (SirTyson) [inline soroban-env-common/env.json:1421]: Remove "expiration" and "bump".

File: soroban-env-common/env.json [hunk 3]
```diff
@@ -1373,52 +1373,52 @@
...
                         },
                         {
-                            "name": "high_expiration_watermark",
+                            "name": "extend_to",
                             "type": "U32Val"
                         }         
                     ],
                     "return": "Void",
-                    "docs": "If expiration for the current contract instance and code (if applicable) is below `low_expiration_watermark` ledgers from the current ledger (inclusive), then bump the expiration to be `high_expiration_watermark` from the current ledger (inclusive)"
+                    "docs": "If expiration for the current contract instance and code (if applicable) is below `threshold` ledgers from the current ledger (inclusive), then bump the expiration to be `extend_to` from the current ledger (inclusive)"
```
Comment (SirTyson) [inline soroban-env-common/env.json:1401]: Remove "expiration" and "bump".

File: soroban-env-host/src/e2e_invoke.rs [hunk 4]
```diff
@@ -78,10 +78,10 @@ pub struct LedgerEntryExpirationChange {
     /// Durability of the entry.    
     pub durability: ContractDataDurability,
     /// Expiration ledger of the old entry.
```
Comment (SirTyson) [inline soroban-env-host/src/e2e_invoke.rs:80]: Remove term expiration from comment.

File: soroban-env-host/src/e2e_invoke.rs [hunk 5]
```diff
@@ -180,8 +180,8 @@ pub fn extract_rent_changes(ledger_changes: &Vec<LedgerEntryChange>) -> Vec<Ledg
                 &entry_change.encoded_new_value,
             ) {
                 // Skip the entry if 1. it is not bumped and 2. the entry size has not increased
-                if expiration_change.old_expiration_ledger
-                    >= expiration_change.new_expiration_ledger
+                if expiration_change.old_live_until_ledger
```
Comment (SirTyson) [inline soroban-env-host/src/e2e_invoke.rs:183]: Rename to `ttl_change`

File: soroban-env-host/src/e2e_invoke.rs [hunk 6]
```diff
@@ -91,7 +91,7 @@ pub fn get_ledger_changes<T: SnapshotSource>(
     budget: &Budget,
     storage: &Storage,
     init_storage_snapshot: &T,
-    init_expiration_entries: ExpirationEntryMap,
+    init_expiration_entries: TtlEntryMap,
```
Comment (SirTyson) [inline soroban-env-host/src/e2e_invoke.rs:94]: `init_ttl_entries`

File: soroban-env-host/src/fees.rs [hunk 7]
```diff
@@ -7,7 +7,7 @@
 /// Rough estimate of the base size of any transaction result in the archives
 /// (independent of the transaction envelope size).
 pub const TX_BASE_RESULT_SIZE: u32 = 300;
-/// Estimate for any `ExpirationEntry` ledger entry
+/// Estimate for any `TtlEntry` ledger entry
 pub const EXPIRATION_ENTRY_SIZE: u32 = 48;
```
Comment (SirTyson) [inline soroban-env-host/src/fees.rs:11]: `TTL_ENTRY_SIZE`

File: soroban-env-host/src/fees.rs [hunk 8]
```diff
@@ -91,9 +91,9 @@ pub struct LedgerEntryRentChange {
     pub new_size_bytes: u32,
     /// Expiration ledger of the entry before it has been modified.
```
Comment (SirTyson) [inline soroban-env-host/src/fees.rs:92]: Remove "expiration" from comments.

File: soroban-env-host/src/host.rs [hunk 9]
```diff
@@ -1692,62 +1692,57 @@ impl VmCallerEnv for Host {
...
+        extend_to: U32Val,
     ) -> Result<Void, HostError> {
         self.check_val_integrity(k)?;
         if matches!(t, StorageType::Instance) {
             return Err(self.err(
                 ScErrorType::Storage,
                 ScErrorCode::InvalidAction,
-                "instance storage should be bumped via `bump_current_contract_instance_and_code` function only",
+                "instance storage should be bumped via `extend_current_contract_instance_and_code` function only",
                 &[],
```
Comment (SirTyson) [inline soroban-env-host/src/host.rs:1709]: "should be extended"

File: soroban-env-host/src/native_contract/token/allowance.rs [hunk 10]
```diff
@@ -37,7 +37,7 @@ pub fn write_allowance(
     // validates the expiration and then returns the ledger seq
     // The expiration can be less than ledger seq if clearing an allowance
     let ledger_seq = e.with_ledger_info(|li| {
-        if expiration > e.max_expiration_ledger()? {
+        if expiration > e.max_live_until_ledger()? {
```
Comment (SirTyson) [inline soroban-env-host/src/native_contract/token/allowance.rs:40]: Rename `expiration` and remove from comment.

File: soroban-env-host/src/native_contract/token/allowance.rs [hunk 11]
```diff
@@ -67,14 +67,14 @@ pub fn write_allowance(
             let mut updated_allowance: AllowanceValue = allowance.try_into_val(e)?;
             updated_allowance.amount = amount;
 
-            let old_expiration = updated_allowance.expiration_ledger;
-            updated_allowance.expiration_ledger = expiration;
+            let old_expiration = updated_allowance.live_until_ledger;
```
Comment (SirTyson) [inline soroban-env-host/src/native_contract/token/allowance.rs:70]: Rename `old_expiration`

File: soroban-env-host/src/native_contract/token/contract.rs [hunk 12]
```diff
@@ -21,7 +21,7 @@ use super::balance::{
 };
 use super::metadata::{read_name, read_symbol, set_metadata, DECIMAL};
 use super::public_types::{AlphaNum12AssetInfo, AlphaNum4AssetInfo};
-use super::storage_types::{INSTANCE_BUMP_AMOUNT, INSTANCE_LIFETIME_THRESHOLD};
+use super::storage_types::{INSTANCE_BUMP_AMOUNT, INSTANCE_TTL_THRESHOLD};
```
Comment (SirTyson) [inline soroban-env-host/src/native_contract/token/contract.rs:24]: Rename to `INSTANCE_EXTEND_AMOUNT`

File: soroban-env-host/src/storage.rs [hunk 13]
```diff
@@ -372,11 +370,10 @@ impl Storage {
             ));
         }
 
-        let new_expiration = host.with_ledger_info(|li| {
-            Ok(li.sequence_number.saturating_add(high_expiration_watermark))
-        })?;
+        let new_expiration =
```
Comment (SirTyson) [inline soroban-env-host/src/storage.rs:373]: `new_live_until`

File: soroban-test-wasms/wasm-workspace/write_upgrade_bytes/src/lib.rs [hunk 14]
```diff
@@ -2,7 +2,7 @@
 use soroban_sdk::{contract, contractimpl, Env, Bytes, BytesN};
 
 pub(crate) const BUMP_AMOUNT: u32 = 518400; // 30 days
```
Comment (SirTyson) [inline soroban-test-wasms/wasm-workspace/write_upgrade_bytes/src/lib.rs:4]: Rename to `EXTEND_AMOUNT`

**Review by SirTyson — COMMENTED** [review]
Some nit picks for out of date terms "expiration" and "bump".

File: soroban-env-common/env.json (hunk 1 above)
Comment (sisuresh) [reply soroban-env-common/env.json]: Fixed

File: soroban-env-common/env.json (hunk 2 above)
Comment (sisuresh) [reply soroban-env-common/env.json]: Fixed

File: soroban-env-common/env.json (hunk 3 above)
Comment (sisuresh) [reply soroban-env-common/env.json]: Fixed

File: soroban-env-host/src/e2e_invoke.rs (hunk 4 above)
Comment (sisuresh) [reply soroban-env-host/src/e2e_invoke.rs]: Fixed

File: soroban-env-host/src/e2e_invoke.rs (hunk 5 above)
Comment (sisuresh) [reply soroban-env-host/src/e2e_invoke.rs]: Fixed

File: soroban-env-host/src/e2e_invoke.rs (hunk 6 above)
Comment (sisuresh) [reply soroban-env-host/src/e2e_invoke.rs]: Fixed

File: soroban-env-host/src/fees.rs (hunk 7 above)
Comment (sisuresh) [reply soroban-env-host/src/fees.rs]: Fixed

File: soroban-env-host/src/fees.rs (hunk 8 above)
Comment (sisuresh) [reply soroban-env-host/src/fees.rs]: Done

File: soroban-env-host/src/host.rs (hunk 9 above)
Comment (sisuresh) [reply soroban-env-host/src/host.rs:1709]: Fixed

File: soroban-env-host/src/native_contract/token/allowance.rs (hunk 10 above)
Comment (sisuresh) [reply soroban-env-host/src/native_contract/token/allowance.rs]: Fixed

File: soroban-env-host/src/native_contract/token/allowance.rs (hunk 11 above)
Comment (sisuresh) [reply soroban-env-host/src/native_contract/token/allowance.rs]: Fixed

File: soroban-env-host/src/native_contract/token/contract.rs (hunk 12 above)
Comment (sisuresh) [reply soroban-env-host/src/native_contract/token/contract.rs]: Fixed

File: soroban-env-host/src/storage.rs (hunk 13 above)
Comment (sisuresh) [reply soroban-env-host/src/storage.rs]: Fixed

File: soroban-test-wasms/wasm-workspace/write_upgrade_bytes/src/lib.rs (hunk 14 above)
Comment (sisuresh) [reply soroban-test-wasms/wasm-workspace/write_upgrade_bytes/src/lib.rs]: Fixed

**Review by graydon — APPROVED** [review]

**Pull request merged**